src/Compilers/CSharp/Test/Semantic/Semantics/RecordStructTests.cs

```diff
@@ -0,0 +1,1280 @@
+}");
+            comp.VerifyDiagnostics(
+                // (3,21): error CS0171: Field 'S.E' must be fully assigned before control is returned to the caller
+                // public data struct S(int x, int y)
+                Diagnostic(ErrorCode.ERR_UnassignedThis, "(int x, int y)").WithArguments("S.E").WithLocation(3, 21),
+                // (3,21): error CS0171: Field 'S.Z' must be fully assigned before control is returned to the caller
+                // public data struct S(int x, int y)
+                Diagnostic(ErrorCode.ERR_UnassignedThis, "(int x, int y)").WithArguments("S.Z").WithLocation(3, 21),
+                // (5,25): warning CS0067: The event 'S.E' is never used
+                //     public event Action E;
+                Diagnostic(ErrorCode.WRN_UnreferencedEvent, "E").WithArguments("S.E").WithLocation(5, 25)
+            );
+
+            var s = comp.GlobalNamespace.GetTypeMember("S");
+            var clone = s.GetMethod(WellKnownMemberNames.CloneMethodName);
+            Assert.Equal(0, clone.Arity);
+            Assert.Equal(0, clone.ParameterCount);
+            Assert.Equal(s, clone.ReturnType);
+
+            var ctor = (MethodSymbol)s.GetMembers(".ctor")[1];
+            Assert.Equal(1, ctor.ParameterCount);
+            Assert.True(ctor.Parameters[0].Type.Equals(s, TypeCompareKind.ConsiderEverything));
+        }
+
+        [Fact(Skip = "PROTOTYPE(record-structs)")]
+        public void RecordClone4_1()
+        {
+            var comp = CreateCompilation(@"
+using System;
+public data struct S(int x, int y)
+{
+    public event Action E = null;
+    public int Z = 0;
+}");
+            comp.VerifyDiagnostics(
+                // (5,25): error CS0573: 'S': cannot have instance property or field initializers in structs
+                //     public event Action E = null;
+                Diagnostic(ErrorCode.ERR_FieldInitializerInStruct, "E").WithArguments("S").WithLocation(5, 25),
+                // (5,25): warning CS0414: The field 'S.E' is assigned but its value is never used
+                //     public event Action E = null;
+                Diagnostic(ErrorCode.WRN_UnreferencedFieldAssg, "E").WithArguments("S.E").WithLocation(5, 25),
+                // (6,16): error CS0573: 'S': cannot have instance property or field initializers in structs
+                //     public int Z = 0;
+                Diagnostic(ErrorCode.ERR_FieldInitializerInStruct, "Z").WithArguments("S").WithLocation(6, 16)
+                );
+        }
+
+        [Fact]
+        public void RecordStructLanguageVersion()
+        {
+            var src1 = @"
+struct Point(int x, int y);
+";
+            var src2 = @"
+record struct Point { }
+";
+            var src3 = @"
+record struct Point(int x, int y);
+";
+
+            var comp = CreateCompilation(src1, parseOptions: TestOptions.Regular9, options: TestOptions.ReleaseDll);
+            comp.VerifyDiagnostics(
+                // (2,13): error CS1514: { expected
```

Comment:
I'm starting to wonder whether such type declarations should instead be treated as records with a missing 'record' keyword. Not asking you to do it in this PR but I'm curious what you think.

Reply:
It's not obvious. If someone is making the mistake of thinking that  primary constructors are generally a thing, then they really don't want to add "record" to fix this. I'll add a note for follow-up discussion.

---
In reply to: [588779583](https://github.com/dotnet/roslyn/pull/51494#discussion_r588779583) [](ancestors = 588779583)